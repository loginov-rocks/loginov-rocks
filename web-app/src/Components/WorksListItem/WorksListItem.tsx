import { GitHubRepo } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

import { GitHubWorkDetails } from 'Components/GitHubWorkDetails';
import { Link } from 'Components/Link';

interface Props {
  gitHubRepos?: GitHubRepo[];
  work: string;
}

export const WorksListItem: React.FunctionComponent<Props> = ({ gitHubRepos, work }) => {
  const gitHubRepo = gitHubRepos ? gitHubRepos.find((repo) => repo.name === work) : undefined;

  let title = <strong>{work}</strong>;
  if (gitHubRepo && gitHubRepo.homepageUrl) {
    title = <Link href={gitHubRepo.homepageUrl}>{title}</Link>;
  }
  if (gitHubRepo && gitHubRepo.isArchived) {
    title = <s title="Archived">{title}</s>;
  }

  return (
    <li>

      {title}

      {gitHubRepo && (
        <>
          {' '}
          /
          {' '}
          <Link href={gitHubRepo.url}>GitHub</Link>
        </>
      )}

      {gitHubRepo && (
        <>
          <br />
          <GitHubWorkDetails repo={gitHubRepo} />
        </>
      )}

    </li>
  );
};

WorksListItem.defaultProps = {
  gitHubRepos: [],
};
