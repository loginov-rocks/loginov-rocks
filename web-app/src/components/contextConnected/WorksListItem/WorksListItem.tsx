import * as React from 'react';

import { GitHubWorkDetails } from 'components/shared/GitHubWorkDetails';
import { ExternalLink } from 'components/shared/ExternalLink';
import { useGitHubContext } from 'contexts/GitHubContext';

interface Props {
  gitHubId: string;
}

export const WorksListItem: React.FC<Props> = ({ gitHubId }) => {
  const gitHubData = useGitHubContext();
  const gitHubRepo = gitHubData && gitHubData.repos
    ? gitHubData.repos.find((repo) => repo.name === gitHubId) : undefined;

  let title = <strong>{gitHubId}</strong>;
  if (gitHubRepo && gitHubRepo.homepageUrl) {
    title = <ExternalLink href={gitHubRepo.homepageUrl}>{title}</ExternalLink>;
  }
  if (gitHubRepo && gitHubRepo.isArchived) {
    title = <s title="Archived">{title}</s>;
  }

  return (
    <>

      {title}

      {gitHubRepo && (
        <>
          {' '}
          /
          {' '}
          <ExternalLink href={gitHubRepo.url}>GitHub</ExternalLink>
        </>
      )}

      {gitHubRepo && (
        <>
          <br />
          <GitHubWorkDetails repo={gitHubRepo} />
        </>
      )}

    </>
  );
};
