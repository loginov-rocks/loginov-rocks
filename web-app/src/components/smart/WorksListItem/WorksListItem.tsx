import * as React from 'react';

import { GitHubWorkDetails } from 'components/dumb/GitHubWorkDetails';
import { Link } from 'components/dumb/Link';
import { useGitHubContext } from 'contexts/GitHubContext';

interface Props {
  gitHubId: string;
}

export const WorksListItem: React.FC<Props> = ({ gitHubId }) => {
  const gitHubData = useGitHubContext();
  const gitHubRepo = gitHubData && gitHubData.repos
    ? gitHubData.repos.find((repo) => repo.name === gitHubId) : undefined;

  let title = <strong>{gitHubId}</strong>;
  if (gitHubRepo && gitHubRepo.homepageUrl) {
    title = <Link href={gitHubRepo.homepageUrl}>{title}</Link>;
  }
  if (gitHubRepo && gitHubRepo.isArchived) {
    title = <s title="Archived">{title}</s>;
  }

  return (
    <>

      {title}

      {gitHubRepo && (
        <>
          {' '}
          /
          {' '}
          <Link href={gitHubRepo.url}>GitHub</Link>
        </>
      )}

      {gitHubRepo && (
        <>
          <br />
          <GitHubWorkDetails repo={gitHubRepo} />
        </>
      )}

    </>
  );
};
