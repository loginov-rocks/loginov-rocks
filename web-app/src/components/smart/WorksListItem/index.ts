import { WorksListItem } from './WorksListItem';

export { WorksListItem };
