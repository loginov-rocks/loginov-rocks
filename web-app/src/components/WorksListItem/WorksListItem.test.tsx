/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { WorksListItem } from './WorksListItem';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const gitHubMockData = require('@loginov-rocks/loginov-rocks-shared/src/GitHub/__fixtures__/data.json');

it('matches snapshot', () => {
  const wrapper = shallow(<WorksListItem work="Some work" />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot with GitHub repo', () => {
  const wrapper = shallow(
    <WorksListItem
      gitHubRepos={gitHubMockData.repos}
      work="Angular-Gulp-Boilerplate"
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
