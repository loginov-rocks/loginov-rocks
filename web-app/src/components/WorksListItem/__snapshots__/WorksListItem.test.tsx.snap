// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li>
  <strong>
    Some work
  </strong>
</li>
`;

exports[`matches snapshot with GitHub repo 1`] = `
<li>
  <Link
    href="https://www.npmjs.com/package/angular-gulp-boilerplate"
  >
    <strong>
      Angular-Gulp-Boilerplate
    </strong>
  </Link>
   
  /
   
  <Link
    href="https://github.com/loginov-rocks/Angular-Gulp-Boilerplate"
  >
    GitHub
  </Link>
  <br />
  <GitHubWorkDetails
    repo={
      Object {
        "description": "Clean but full featured AngularJS boilerplate using Gulp workflow and best practices",
        "homepageUrl": "https://www.npmjs.com/package/angular-gulp-boilerplate",
        "isArchived": false,
        "language": "JavaScript",
        "latestTag": "v1.0.10",
        "name": "Angular-Gulp-Boilerplate",
        "stars": 29,
        "updatedAt": 1614276648000,
        "url": "https://github.com/loginov-rocks/Angular-Gulp-Boilerplate",
      }
    }
  />
</li>
`;
